File: test-npm-pip-pkg/ts/tests/test-platform.ts
import { pathToFileURL } from 'url';
import { client, runTestSection, logResult } from "../utils/utils.js";

async function testPlatform() {
  console.log("\n🚀 Testing Platform, Organizations, and Teams...");

  let orgId: string | null = null;
  let teamId: string | null = null;
  let userId: string | null = null;

  // 1. Users
  await runTestSection("platform.getMe", async () => {
    const me = await client.platform.getMe();
    userId = me._id || me.id;
    logResult("Me", me);
  });

  await runTestSection("platform.listUsers", async () => {
    const users = await client.platform.listUsers({ limit: 5 });
    logResult("Users Count", users.data.length);
  });

  // 2. Organizations
  await runTestSection("platform.listOrgs", async () => {
    const orgs = await client.platform.listOrgs({ limit: 5 });
    logResult("Orgs Count", orgs.data.length);
  });

  await runTestSection("organizations.listAll", async () => {
    const orgs = await client.organizations.listAll();
    logResult("All Orgs Count", orgs.length);
  });

  // 3. Teams
  await runTestSection("teams.create", async () => {
    const team = await client.teams.create({
      name: `SDK Test Team ${Date.now()}`,
      description: "Created by automated test"
    });
    teamId = team._id || team.id;
    logResult("Team Created", teamId);
  });

  if (teamId) {
    await runTestSection("teams.update", () =>
      client.teams.update(teamId!, { description: "Updated description" })
    );

    await runTestSection("teams.listMy", async () => {
      const teams = await client.teams.listMy();
      logResult("My Teams Count", teams.length);
    });

    if (userId) {
      await runTestSection("teams.addUsers", () =>
        client.teams.addUsers({ team_id: teamId!, user_ids: [userId!] })
      );
    }

    await runTestSection("teams.delete", () => client.teams.delete(teamId!));
  }

  // 4. System / Apps
  await runTestSection("platform.listApps", async () => {
    const apps = await client.platform.listApps();
    logResult("Apps Count", apps.data?.length || 0);
  });

  await runTestSection("platform.listCredentials", async () => {
    const creds = await client.platform.listCredentials();
    logResult("Credentials Count", creds.length);
  });

  await runTestSection("platform.getMenuSettings", async () => {
    const settings = await client.platform.getMenuSettings();
    logResult("Menu Settings", !!settings);
  });

  await runTestSection("platform.listPermissions", async () => {
    if (userId) {
        const perms = await client.platform.listPermissions(userId!);
        logResult("Permissions Count", perms.length);
    }
  });

  console.log("\n✅ Platform Resource Testing Completed.");
}

if (import.meta.url === pathToFileURL(process.argv[1]).href) {
  testPlatform().catch(console.error);
}

export { testPlatform };

File: test-local-pkg/py/tests/test_platform.py
import time
from utils.utils import client, run_test_section, log_result

def test_platform():
    print("\n🚀 Testing Platform, Organizations, and Teams...")

    state = {"user_id": None, "team_id": None}

    # 1. Users
    def users_ops():
        me = client.platform.get_me()
        state["user_id"] = me.get("_id") or me.get("id")
        log_result("Me", me)

        users = client.platform.list_users(limit=5)
        log_result("Users Count", len(users.get("data", [])))
    run_test_section("Users Ops", users_ops)

    # 2. Organizations
    run_test_section("platform.list_orgs", lambda: log_result("Orgs Count", len(client.platform.list_orgs(limit=5).get("data", []))))

    # 3. Teams
    def teams_ops():
        team = client.teams.create({
            "name": f"SDK Test Team {int(time.time())}",
            "description": "Created by automated test"
        })
        state["team_id"] = team.get("_id") or team.get("id")
        log_result("Team Created", state["team_id"])

        if state["team_id"]:
            client.teams.update(state["team_id"], {"description": "Updated description"})
            log_result("Team Updated", True)

            client.teams.delete(state["team_id"])
            log_result("Team Deleted", True)
    run_test_section("Teams Ops", teams_ops)

    # 4. System / Apps
    run_test_section("platform.list_apps", lambda: log_result("Apps Count", len(client.platform.list_apps().get("data", []))))

    print("\n✅ Platform Resource Testing Completed.")

if __name__ == "__main__":
    test_platform()
